models/utils.py

```diff
@@ -0,0 +1,43 @@
+import operator
+from datetime import datetime
+from dmutils.formats import DATE_FORMAT, DATETIME_FORMAT
+
+
+format_datetime_string_as_date = lambda x: datetime.strptime(x, DATETIME_FORMAT).strftime(DATE_FORMAT)
+remove_username_from_email_address = lambda x: '@{}'.format(x.split('@').pop())
+
+
+def process_collection(rules, collection):
+    if not rules:
+        return collection
+
+    for c in collection:
+        for k, v in rules.items():
+            if c.get(k):
+                c[k] = v(c[k])
+
+
+def return_filtered_collection(rules, collection):
+    if not rules:
+        return collection
+
+    # `contains` would be trickier to include because it switches the argument order
+    ops = {
+        '>': operator.gt,
+        '<': operator.lt,
+        '>=': operator.ge,
+        '<=': operator.le,
+        '==': operator.eq
+    }
+
+    for key, op, to_compare in rules:
+        collection = list(filter(lambda x: ops[op](x[key], to_compare), collection))
```

Comment:
Yeah, I think we're planning to use something like [pandas](http://pandas.pydata.org/) in the (near) future and at that point we can replace this logic.

Definitely agreed -- don't really want to go down this road any further.